examples/stable/test/property/ic_api/json/test/tests.ts

```diff
@@ -190,71 +94,61 @@ export function getTests(): Test {
-                expect(queryResult).toEqual(updateResult);
-            }
+                    // Parse the returned JSON string and verify it matches the original
+                    const parsedReturnedValue = jsonParse(returnedJsonString);
+                    expect(jsonStringify(parsedReturnedValue)).toEqual(
+                        jsonString
+                    );
+                }),
+                executionParams
+            );
         });
 
-        it('should test simple values with property tests', async () => {
-            // Use a smaller number of runs for property tests
+        it('should test jsonStringify and jsonParse with update method', async () => {
+            // Multiply the default number of runs by 100 as requested in the issue
             const executionParams = {
                 ...defaultPropTestParams(),
-                numRuns: 10 * Number(process.env.AZLE_PROPTEST_NUM_RUNS ?? 1)
+                numRuns: 100 * Number(process.env.AZLE_PROPTEST_NUM_RUNS ?? 1)
             };
 
             await fc.assert(
-                fc.asyncProperty(simpleTestArb, async (value) => {
+                fc.asyncProperty(mixedObjectArb, async (value) => {
                     const actor = await getCanisterActor<Actor>('canister');
 
                     // Convert the value to a JSON string using jsonStringify
                     const jsonString = jsonStringify(value);
 
-                    // Call the canister methods and test both query and update
-                    const queryResult =
-                        await actor.processJsonQuery(jsonString);
-                    const updateResult =
+                    // Call the canister method that parses and then stringifies again
+                    const returnedJsonString =
                         await actor.processJsonUpdate(jsonString);
 
-                    // The returned strings should match the original
-                    expect(queryResult).toEqual(jsonString);
-                    expect(updateResult).toEqual(jsonString);
+                    // The returned string should match the original
+                    expect(returnedJsonString).toEqual(jsonString);
 
-                    // Parse the returned JSON strings and verify they match the original
-                    const parsedQueryResult = jsonParse(queryResult);
-                    const parsedUpdateResult = jsonParse(updateResult);
-
-                    // Verify the objects can be re-stringified and still match
-                    expect(jsonStringify(parsedQueryResult)).toEqual(
-                        jsonString
-                    );
-                    expect(jsonStringify(parsedUpdateResult)).toEqual(
+                    // Parse the returned JSON string and verify it matches the original
+                    const parsedReturnedValue = jsonParse(returnedJsonString);
+                    expect(jsonStringify(parsedReturnedValue)).toEqual(
```

Comment:
But you should not stringify the parsed return value, you should directly compare it with the original arbitrary value.